### ISSUE TITLE
ClassKit entitlement support for Xcode 9.4

### ISSUE THREAD

**Issue by dalexsoto** [xamarin/xamarin-macios#3911]
### Steps to Reproduce

Xcode 9.4 introduced ClassKit entitlement 

From `Entitlements.plist`:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.developer.ClassKit-environment</key>
	<string>development</string>
</dict>
</plist>
```

According to [documentation](https://developer.apple.com/documentation/classkit/testing_your_app_during_development?language=objc) (see **Optionally Test in Production Mode** section) it has two possible values:

* development
* production

### Expected Behavior

Have support for `com.apple.developer.ClassKit-environment` entitlement.

## Workaround

1. Manually add `com.apple.developer.ClassKit-environment` entitlement to your Entitlements.plist file manually as shown above in the xml code section.
2. Enable the automatic provisioning profiles as explained here https://blog.xamarin.com/even-easier-ios-app-provisioning-visual-studio-mac/
3. Go to [Apple developer portal](https://developer.apple.com) and head to **Certificates, Identifiers & Profiles** section.
4. Inside the portal click on `Identifiers > Apps IDs` section.
<img width="249" alt="step4" src="https://user-images.githubusercontent.com/204671/38563417-739baea6-3ca2-11e8-875b-88358c5742f6.png">
5. Look for your App bundle Id, it should be prefixed with **VS** plus the bundle identifier inside your info.plist (In our example it is `VS com xamarin GreatPlays`) and click the edit button in the lower right. 
<img width="738" alt="step5" src="https://user-images.githubusercontent.com/204671/38563880-8d4c4238-3ca3-11e8-9d85-fff0b742e37b.png">
6. Enable `ClassKit` and hit `Done`.
<img width="706" alt="step6" src="https://user-images.githubusercontent.com/204671/38564070-f804c212-3ca3-11e8-99e7-53c93eb2c83a.png">
7. Now you should be able to go back to VS4Mac and deploy to your device.

**Comment by olegoid** [comment]
@dalexsoto 
https://github.com/xamarin/md-addins/pull/3124

**Comment by olegoid** [comment]
The widget will be released in 15.8. Closing the issue for now. All further related progress will be tracked in this VSTS task:

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/604982
